design/one-pager-stack-relationship-labels.md

```diff
@@ -0,0 +1,166 @@
+    app.kubernetes.io/managed-by: stack-manager
+...
+```
+
+### Example wordpress `resource.yaml`
+
+```yaml
+apiVersion: wordpress.samples.stacks.crossplane.io/v1alpha1
+kind: WordpressInstance
+metadata:
+  annotations:
+    kubectl.kubernetes.io/last-applied-configuration: |
+      {
+        "apiVersion":"wordpress.samples.stacks.crossplane.io/v1alpha1",
+        "kind":"WordpressInstance",
+        "metadata":{
+          "annotations":{},
+          "name":"my-wordpressinstance",
+          "namespace":"app-project1-dev"
+        }
+      }
+  creationTimestamp: "2019-10-23T23:47:40Z"
+  generation: 1
+  name: my-wordpressinstance
+  namespace: app-project1-dev
+  resourceVersion: "6445"
+  selfLink: /apis/wordpress.samples.stacks.crossplane.io/v1alpha1/namespaces/app-project1-dev/wordpressinstances/my-wordpressinstance
+  uid: f2d13a15-1f9b-40a7-a173-a40abefa61bf
+  labels:
+    core.crossplane.io/parent-kind: "customresourcedefinition.apiextensions.k8s.io"
+    core.crossplane.io/parent-name: "sample-stack-wordpress"
+    core.crossplane.io/parent-namespace: "app-project1-dev"
+    app.kubernetes.io/managed-by: stack-manager
+...
+```
+
+### Example wordpress `kubernetesapplication.yaml`
+
+```yaml
+apiVersion: workload.crossplane.io/v1alpha1
+kind: KubernetesApplication
+metadata:
+  name: wordpress-app-wordpress
+  labels:
+    stack: sample-stack-wordpress
+    core.crossplane.io/parent-kind: "wordpressinstances.wordpress.samples.stacks.crossplane.io"
+    core.crossplane.io/parent-name: "my-wordpressinstance"
+    core.crossplane.io/parent-namespace: "app-project1-dev"
+    app.kubernetes.io/managed-by: stack-manager
+...
+```
+
+### Example wordpress `kubernetescluster.yaml`
+
+```yaml
+apiVersion: compute.crossplane.io/v1alpha1
+kind: KubernetesCluster
+metadata:
+  name: wordpress-cluster-wordpress
+  labels:
+    stack: sample-stack-wordpress
+    core.crossplane.io/parent-kind: "wordpressinstances.wordpress.samples.stacks.crossplane.io"
+    core.crossplane.io/parent-name: "my-wordpressinstance"
+    core.crossplane.io/parent-namespace: "app-project1-dev"
+    app.kubernetes.io/managed-by: stack-manager
+...
+```
+
+### Example wordpress `mysqlinstance.yaml`
+
+```yaml
+apiVersion: database.crossplane.io/v1alpha1
+kind: MySQLInstance
+metadata:
+  name: wordpress-mysql-wordpress
+  labels:
+    stack: sample-stack-wordpress
+    core.crossplane.io/parent-kind: "wordpressinstances.wordpress.samples.stacks.crossplane.io"
+    core.crossplane.io/parent-name: "my-wordpressinstance"
+    core.crossplane.io/parent-namespace: "app-project1-dev"
+    app.kubernetes.io/managed-by: stack-manager
+...
+```
+
+As per the above file example snippets, as long as the labels are defined on children objects,
+then we can use that label to look up all children of a given parent.
+
+Using `kubectl` for example:
+
+```bash
+kubectl get kubernetesapplication -l core.crossplane.io/parent-name=my-wordpressinstance
```

Comment:
I think this is glossing over a particular detail about children types.  This query is using information that appears to be already known that they are looking for a child of type `kubernetesapplication`.  Is there a good way to do this generically, when you don't know what types the children will be?  How would an entity query for all children of a given parent, when they don't know what their types may be?

another thought, queries may probably need to use more information from the parent's labels, e.g. specify the parent's kind and namespace too.  `parent-name` isn't specific enough by itself since many resources in k8s can all have the same name.

Reply:
I was kind of on the fence including this line at all since it really isn't a complete accurate call and was really just a very abstract thought about how you *might* leverage these labels.

I think I would prefer to remove it all together

Reply:
> Is there a good way to do this generically, when you don't know what types the children will be? 

I don't believe so.

For example, `kubectl get all -lapp=crossplane -A`, would only list the types in the "all" category, which does not include KubernetesApplication.

If Crossplane types included a categories (`//+kubebuilder:categories=crossplane`), then `kubectl get crossplane,all -lapp=crossplane -A` would be possible.  Multiple categories can be provided in this way (`//+kubebuilder:categories=all,crossplane`).

Reply:
removing this chunk from the one-pager